app/views/postcodes/_postcode_lookup.html.haml

```diff
@@ -9,7 +9,7 @@
     = hidden_field_tag('constituency_id', @constituency.graph_id) if @constituency
     = hidden_field_tag('person_id', @person.graph_id) if @person
     = label_tag 'input', t('.search_by_postcode')
-    = text_field_tag(:postcode, @postcode, value: flash[:postcode], maxlength: 8, pattern:'[0-9a-zA-Z ]{5,}', id:'input')
+    = text_field_tag(:postcode, @postcode, value: flash[:postcode], maxlength: 8, pattern:'[A-Za-z]{1,2}[0-9Rr][0-9A-Za-z] ?[0-9][ABD-HJLNP-UW-Zabd-hjlnp-uw-z]{2}', id:'input')
```

Comment:
It might also be worth exposing this from a helper to allow for reuse elsewhere.